packages/@lwc/engine-core/src/framework/wiring.ts

```diff
@@ -339,11 +340,14 @@ export function installWireAdapters(vm: VM) {
             const hasDynamicParams = wireDef.dynamic.length > 0;
             ArrayPush.call(wiredConnecting, () => {
                 connector.connect();
-                if (hasDynamicParams) {
-                    Promise.resolve().then(computeConfigAndUpdate);
-                } else {
-                    computeConfigAndUpdate();
+                if (!featureFlags.ENABLE_WIRE_SYNC_EMIT) {
+                    if (hasDynamicParams) {
+                        Promise.resolve().then(computeConfigAndUpdate);
+                        return;
+                    }
                 }
+
+                computeConfigAndUpdate();
```

Comment:
This should make the code more readable. 

```suggestion
                if (featureFlags.ENABLE_WIRE_SYNC_EMIT) {
                    computeConfigAndUpdate();
                } else {
                    if (hasDynamicParams) {
                        Promise.resolve().then(computeConfigAndUpdate);
                    }
                }
```